server/controllers/payment.js

```diff
@@ -0,0 +1,74 @@
+const asyncHandler = require("express-async-handler");
+const Payment = require("../models/Payment");
+
+
+
+exports.getPayment = asyncHandler(async (req, res, next) => {
+  const {id} = req.params;
+  const payment = await Payment.findById(id);
+  if(payment){
+    res.status(200).json({
+      success: {   
+        payment: payment 
+      }
+    });
+  } else {
+    res.status(404);
+    throw new Error("Pls enter valid payment id");
+  }   
```

Comment:
Could we add some whitespace above this conditional. Also, Could we do a different approach.

```js
if (!payment) {
   . . .
}

return res.status(200).json({  . . . });
```

Reply:
Ditto throughout the controller